src/Symfony/Component/BrowserKit/AbstractBrowser.php

```diff
@@ -366,11 +366,11 @@ public function request(string $method, string $uri, array $parameters = [], arr
 
         $server = array_merge($this->server, $server);
 
-        if (!empty($server['HTTP_HOST']) && null === parse_url($originalUri, \PHP_URL_HOST)) {
+        if (!empty($server['HTTP_HOST']) && !parse_url($originalUri, \PHP_URL_HOST)) {
```

Comment:
What is the point here? Do we also want to cover completely malformed URLs for which `false` is returned?